File: v1/ruby/auth_token.rb
require 'net/http'
require 'json'

class StriimRestClient
  
  AUTH_PATH = '/security/authenticate'

  def initialize(url)
    @url = url
  end
  
  def auth_token(user, pass)
    uri = URI(@url + AUTH_PATH)
    req = Net::HTTP::Post.new(uri) 
    req.set_form_data('username' => user, 'password' => pass )
    res = Net::HTTP.start(uri.hostname, uri.port) do | http |
      http.request(req)
    end

    case res
    when Net::HTTPSuccess, Net::HTTPRedirection
      j = JSON.parse(res.body)
      j['token']
    else
      raise 'Error authenticating'
    end
  end
end
#Uncomment this also if you want direct output.
#client=StriimRestClient.new()
#client.getAuthToken('admin','admin')

File: v1/python/AuthToken.py
#lib imports
import json
import smtplib
import base64
import urllib
import urllib2
#requests lib required
import requests

class AuthToken:

   def authenticate(self, username, password):
       values = { 'username': username ,'password':password}
       data = urllib.urlencode(values)
       theToken = None
       try:
           url = "http://localhost:9080/security/authenticate"
           req = urllib2.Request(url, data)
           response = urllib2.urlopen(req)
           result = response.read()
           tokenjson = json.loads(result)
           theToken = tokenjson["token"]
       except:
           print "ERROR: Incorrect webaction username/password for authentication"
       finally:
           return theToken

if __name__ == '__main__':
    authToken = AuthToken()
    print authToken.authenticate('admin','admin')
